src/content/en/updates/2018/07/watch-video-using-picture-in-picture.md

```diff
@@ -0,0 +1,277 @@
+possible to change state programmatically in JavaScript.
+
+### Exit Picture-in-Picture {: #exit-pip }
+
+But why stop there? Let's make our button toggle entering and exiting
+Picture-in-Picture. We first have to check if the read-only object
+`document.pictureInPictureElement` is our video element. If it isn’t, we
+send a request to enter Picture-in-Picture as above. Otherwise, we ask to leave
+by calling `document.exitPictureInPicture()`, which means the video will appear
+back in the original tab. Note that this method also returns a promise.
+
+    ...
+    try {
+      if (videoElement !== document.pictureInPictureElement) {
+        await videoElement.requestPictureInPicture();
+      } else {
+        await document.exitPictureInPicture();
+      }
+    }
+    ...
+
+### Listen to Picture-in-Picture events {: #listen-pip-events }
+
+Operating systems usually restrict Picture-in-Picture to one window, so
+Chrome's implementation follows this pattern. This means users can only play
+one Picture-in-Picture video at a time. You should expect users to exit
+Picture-in-Picture even when you didn't ask for it.
+
+The new `enterpictureinpicture` and `leavepictureinpicture` event handlers let
+us tailor the experience for users. It could be anything from browsing a
+catalog of videos, to surfacing a livestream chat.
+
+    videoElement.addEventListener('enterpictureinpicture', function(event) {
+      // Video entered Picture-in-Picture.
+    });
+
+    videoElement.addEventListener('leavepictureinpicture', function(event) {
+      // Video left Picture-in-Picture.
+      // User may have played a Picture-in-Picture video from a different page.
+    });
+
+### Get Picture-in-Picture window size
+
+If you want to adjust the video quality when the video enters and leaves
+Picture-in-Picture, you need to know the Picture-in-Picture window size and be
+notified if a user manually resizes the window.
+
+The example below shows how to get the width and height of the
+Picture-in-Picture window when it is created or resized.
+
+    const pipWindow = await videoElement.requestPictureInPicture();
+    updateVideoQuality(pipWindow);
+
+    pipWindow.addEventListener('resize', function(event) {
+      const pipWindow = event.target;
+      updateVideoQuality(pipWindow);
+    });
+
+    function updateVideoQuality(pipWindow) {
+      console.log('Picture-in-Picture window width is ' + pipWindow.width);
+      console.log('Picture-in-Picture window height is ' + pipWindow.height);
+      // TODO: Change video quality based on Picture-in-Picture window size.
+    }
+
+I’d suggest you don’t hook directly to the resize event as each small change
+made to the Picture-in-Picture window size will fire a separate event that may
+cause performance issues if you’re doing an expensive operation at each resize.
+I’d recommend using common techniques such as [throttling and debouncing] to
+address this problem.
+
+### Feature support {: #feature-support }
```

Comment:
I put it at the end of my article because as a web developer It want to learn about the general shape of the API first before all the details I'll need eventually if I decide to use it. That's my reasoning.